flume/datadog_checks/flume/data/metrics.yaml

```diff
@@ -0,0 +1,92 @@
+jmx_metrics:
+    - include:
+        domain: org.apache.flume.channel
+        attribute:
+          ChannelCapacity:
+            metric_type: gauge
+            alias: flume.channel.capacity
+          ChannelFillPercentage:
+            metric_type: gauge
+            alias: flume.channel.fill_percentage
+          ChannelSize:
+            metric_type: gauge
+            alias: flume.channel.size
+          EventPutAttemptCount:
+            metric_type: gauge
+            alias: flume.channel.event_put_attempt_count
```

Comment:
IIUC, this and many other metrics values are monotonically increasing. For exemple here, the value seems to represent the total number of "Event Put Attempts" since flume started. In such a case we usually set the metric_type to either `monotonic_count` (and on Datadog you can graph the count of new events using a bar chart for exemple) or with `rate` (and on Datadog, the timeseries will represent how many new put events per seconds are attempted).

As of now, graphs in Datadog for those metrics will be monotonically increasing which requires user to apply custom transformations (like per_second modifier) to extract a meaning from the metric.

This can stay as is, but I think it's worth mentioning if you want to revisit the metric types.

Reply:
Small tip as well in case you do reconsider some of the metric types, you'll have to change the metadata.csv file but in a different way than what you might expect.
The types defined in the metrics.yaml file are the submission types, (i.e the agent performs calculations on the values before sending them to Datadog), not the in-app types: https://docs.datadoghq.com/developers/metrics/types/?tab=count

The in-app types are defined in metadata.csv, and the usual mapping is as follow:

|submission type | in-app type|
|-----------------|------------|
|gauge|gauge|
|monotonic_count|count|
|rate|gauge|

Reply:
Hey @FlorianVeaux  thank you for the useful notes!
As per your instructions I've revised the prior metric types. 

My apologies for the delay on this- while reviewing the flume documentation on the exposed metrics I noticed Flume can act as a kind of Kafka Consumer/Producer. 
When doing so, there are additional metrics exposed which I had not encountered on the first draft of this integration. 

As such I've updated my testing environment to include a Zookeeper/ Kafka Broker service and my flume Agent to include a Kafka Source/Kaka Channel/Kafka Sink. 

The integration now collects **all** possible Flume metrics documented in the official Apache flume metrics docs. 

I've also updated my E2E tests to account for the additional metrics now collected.

Finally I've updated the integration documentation to link to the Flume Component (Sources/Channels/Sinks) metric documentation- as metrics exposed by Flume are based on which type of each component a user selects. 

Please let me know what you think 👍 

As a final note- 
It seems merging is blocked here because of the Codecov report- it looks like this doesn't account for the E2E tests which are run (As seen in the AZ pipeline). 
Could you possibly merge with admin rights?